PowerEditor/src/Parameters.cpp

```diff
@@ -5861,6 +5870,13 @@ void NppParameters::createXmlTreeFromGUIParams()
 		GUIConfigElement->SetAttribute(TEXT("onAnotherView"), _nppGUI._smartHiliteOnAnotherView ? TEXT("yes") : TEXT("no"));
 	}
 
+	// <GUIConfig name = "commandLineInterpreter">< / GUIConfig>

```

Comment:
> This setting is intentionally not accessible in the Preferences dialogue,
because changing this from cmd (the default value) to something else will most likely require some preparation, like adjusting PATH in the Registry, or creating a hard link in a suitable directory, or even writing a small utility acting as a bridge between NPP and the targeted shell. 

Good point. In this case, I think it'll be even more cautious to remove this block. So if user needs this feature, he/she adds it manually.

Reply:
Good idea. I had not thought about the possibility to make it a ‘read-only’ option.

Code block removed.

Reply:
@donho 
Oops, that doesn't work, because the new `config.xml` will no longer contain the _commandLineInterpreter_ setting. Perhaps, something like this:
```
	// <GUIConfig name = "commandLineInterpreter">< / GUIConfig>
	if (_nppGUI._commandLineInterpreter.compare(TEXT("cmd")))
	{
		TiXmlElement *GUIConfigElement = (newGUIRoot->InsertEndChild(TiXmlElement(TEXT("GUIConfig"))))->ToElement();
		GUIConfigElement->SetAttribute(TEXT("name"), TEXT("commandLineInterpreter"));
		GUIConfigElement->InsertEndChild(TiXmlText(_nppGUI._commandLineInterpreter.c_str()));
	}
```
Only add the _commandLineInterpreter_ setting if its value does not equal "cmd"?

Reply:
I like it better when the block is present even if it says "cmd".  Then, a power user that wants to change it doesn't have to look up the syntax, they just change "cmd" to what they want, and they are up and running with it.  Easy.  My 2c.